state/executor.go

```diff
@@ -132,17 +132,13 @@ func (e *BlockExecutor) CreateBlock(height uint64, lastCommit *types.Commit, las
 	block.SignedHeader.Header.LastCommitHash = e.getLastCommitHash(lastCommit, &block.SignedHeader.Header)
 	block.SignedHeader.Header.LastHeaderHash = lastHeaderHash
 	block.SignedHeader.Header.AggregatorsHash = state.Validators.Hash()
+	block.SignedHeader.Header.NextAggregatorsHash = state.NextValidators.Hash()
```

Comment:
not sure if this will work. the `NextAggregatorsHash` should reflect any changes in the validator set, which is only known after the `UpdateState` (in the manager.go) is called. could you write a test that demonstrates this? there should be something available in the full_node_integration_test which has multiple validators and validator updates. you may have to move the UpdateState call before the getCommit so that you can assign the updated validator set hash.

Reply:
Added a test case here: 
https://github.com/rollkit/rollkit/pull/844/commits/942c5e385d09c6f8e9524b6ead4150fcc741e729

but moving `updateState` before `getCommit` is turning out to be tricky because of the conditional due to `pendingBlock`.

Simply moving the `updateState` and `applyBlock` calls before `getCommit` throws the test into a infinite loop.

Going to investigate how to fix this. Maybe we can pass `nextAggregatorsHash` to `createBlock` which already accepts `lastHeadersHash`.

Reply:
Working on pulling in changes from #1200 to resolve the updated validator set change.

Reply:
Done.